### PR TITLE
[Issue 2648] Add signout endpoint

### PR DESCRIPTION
**Pull request by babebe**
## Summary
Fixes #{[2648](https://github.com/HHS/simpler-grants-gov/issues/2648)}

## Changes proposed.

Added `/token/logout` endpoint 
Added `LogoutResponseSchema` response schema
Added success and invalid token tests

### CODE REVIEW

File: api/tests/src/api/users/test_user_route_token.py [hunk 1]
```diff
@@ -25,3 +28,30 @@ def test_post_user_route_token_400(client, api_auth_token):
...
+    enable_factory_create, client, db_session, api_auth_token
+):
+    user = UserFactory.create()
+    token, _ = create_jwt_for_user(user, db_session)
+    db_session.commit()
+
+    resp = client.post("v1/users/token/logout", headers={"X-SGG-Token": token})
+
+    assert resp.status_code == 200
+
```
Comment (chouinar) [inline api/tests/src/api/users/test_user_route_token.py:43]: I'd check that the record in the DB is updated afterwards.

 Something like:
```
token, user_token_session = create_jwt_for_user(user, db_session)
...
db_session.refresh(user_token_session)
assert user_token_session.is_valid is False
```

File: api/src/api/users/user_routes.py [hunk 2]
```diff
@@ -34,3 +39,26 @@ def user_token(x_oauth_login_gov: dict) -> response.ApiResponse:
...
+    user_token_session: UserTokenSession = api_jwt_auth.current_user  # type: ignore
+    with db_session.begin():
+        user_token_session.is_valid = False
+        db_session.add(user_token_session)
+
+    logger.info(
+        "Logged out a user",
+        extra={
+            "user_token_session.user_id": str(user_token_session.user_id),
+        },
```
Comment (chouinar) [inline api/src/api/users/user_routes.py:61]: I'd add the token user_token_session.token_id as well so we know which token was logged out.

File: api/tests/src/api/users/test_user_route_token.py (hunk 1 above)
Comment (babebe) [reply api/tests/src/api/users/test_user_route_token.py:43]: Added

File: api/src/api/users/user_routes.py (hunk 2 above)
Comment (babebe) [reply api/src/api/users/user_routes.py:61]: Added to log

**Review by chouinar — APPROVED** [review]
LGTM!

**Pull request merged**